enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,347 @@
+
+At a high level, the local control plane is responsible for:
+- Setting up certificates for the apiserver and etcd.
+- Running (and cleaning up, on shutdown) the local control plane components.
+- Installing any required component, like Custom Resource Definitions (CRDs)
+    - For Cluster API core the CRDs are stored in `data/data/cluster-api/core-components.yaml`.
+    - Infrastructure providers are expected to store their components in `data/data/cluster-api/<name>-infrastructure-components.yaml`
+- Upon install, the local control plane takes care of modifying any webhook (conversion, admission, validation) to point to the `host:post` combination assigned.
+    - Each controller manager will have its own `host:port` combination assigned.
+    - Certificates are generated and injected in the server, and the client certs in the api-server webhook configuration.
+- For each process that the local control plane manages, a health check (ping to `/healthz`) is required to pass similarly how, when running in a Deployment, a health probe is configured.
+
+#### Manifests
+
+The Installer will produce the CAPI manifests as part of the `manifests` target, writing them to a new
+`cluster-api` directory alongside the existing `manifests` and `openshift` directories:
+
+```shell=
+$ ./openshift-install create manifests --dir install-dir
+INFO Credentials loaded from the "default" profile in file "~/.aws/credentials" 
+INFO Consuming Install Config from target directory 
+INFO Manifests created in: install-dir/cluster-api, install-dir/manifests and install-dir/openshift 
+$ tree install-dir/cluster-api/
+install-dir/cluster-api/
+├── 00_capi-namespace.yaml
+├── 01_aws-cluster-controller-identity-default.yaml
+├── 01_capi-cluster.yaml
+├── 02_infra-cluster.yaml
+├── 10_inframachine_mycluster-6lxqp-master-0.yaml
+├── 10_inframachine_mycluster-6lxqp-master-1.yaml
+├── 10_inframachine_mycluster-6lxqp-master-2.yaml
+├── 10_inframachine_mycluster-6lxqp-master-bootstrap.yaml
+├── 10_machine_mycluster-6lxqp-master-0.yaml
+├── 10_machine_mycluster-6lxqp-master-1.yaml
+├── 10_machine_mycluster-6lxqp-master-2.yaml
+└── 10_machine_mycluster-6lxqp-master-bootstrap.yaml
+
+1 directory, 12 files
+```
+
+The manifests within this `cluster-api` directory will not be written to the cluster or included in bootstrap ignition.
+In future work, we expect these manifests to be pivoted to the cluster to enable the target cluster to take over managing
+its own infrastructure.
+
+The Cluster API manifests will be generated with `.spec` fields in the [manifest asset target][asset] with `.status`
+updated after cluster creation.
+
+### Risks and Mitigations
+
+While we do not expect these changes to introduce a significant security risk, we are working with product security teams
+to ensure they are aware of the changes and are able to review.
+
+
+### Drawbacks
+
+By depending on CAPI providers whose codebases live in a repository external to the Installer,
```

Comment:
I don't think those logistics have been planned yet. I would assume we'll have our own Red Hat copy of the source for each upstream component we ship, per usual for OSS we depend on, so we can do our own backporting and lifecycle management over a timeline that matches our support obligations.

I read this particular drawback as basically saying "Shipping code from an active upstream project is not as simple as shipping the terraform and MAPI components where we're the only maintainers and contributors."

Reply:
We already have experience in this area from hypershift.

Reply:
> So the plan is not to vendor the code into the installer repo (even to build as separate binaries), but to build all of the binaries from their original source repo and then embed the results?

We'll be vendoring the code and build it in-tree for the first iteration; I think (don't want to speak for @patrickdillon, so keep me honest here Patrick!) the point was more that in this world we should strive to upstream features first, and then integrate them downstream rather than doing everything downstream like we've been able to do with Terraform.

Reply:
> I don't think those logistics have been planned yet. I would assume we'll have our own Red Hat copy of the source for each upstream component we ship, per usual for OSS we depend on, so we can do our own backporting and lifecycle management over a timeline that matches our support obligations.

Those forks already exist and are in payload for AWS, Azure, GCP, vSphere, OpenStack and IBM platforms.

> Are we going to need changes in any ART pipelines to support it?

ART can detect downstream image changes and trigger rebuilds, it already supports what we would need to allow multi-stage container builds to pull binaries from other places in the payload. CI on the other hand, does not currently support this, and is something we would need to work on.

Reply:
Added new section (Copying Dependencies from Container Images) on this. @JoelSpeed PTAL